rules/predicates.py

```diff
@@ -64,13 +64,13 @@ def __init__(
         assert callable(fn), "The given predicate is not callable."
         innerfn = fn
         if isinstance(fn, Predicate):
-            fn, num_args, var_args, name = (
+            innerfn, num_args, var_args, name = (
                 fn.fn,
                 fn.num_args,
                 fn.var_args,
                 name or fn.name,
             )
-            innerfn = fn
+            fn = innerfn
```

Comment:
This should have the same effect as the previous code, but avoids a `mypy` error related to type narrowing `fn`.